packages/codemods/src/helpers/styled-components-attributes.ts

```diff
@@ -0,0 +1,278 @@
+    );
+
+    const quasis = keys.reduce((quasis, key, i) => {
+      const prefix = i === 0 ? '' : ';';
+      const isLast = i === keys.length - 1;
+      const element = `${prefix}\n${key}: `;
+
+      quasis.push(j.templateElement({ cooked: element, raw: element }, false));
+      isLast && quasis.push(j.templateElement({ cooked: ';', raw: ';' }, false));
+
+      return quasis;
+    }, [] as TemplateElement[]);
+
+    return { quasis, expressions };
+  }
+
+  return { quasis: [], expressions: [] };
+};
+
+const createComponentWrapper = ({
+  j,
+  expression,
+  attrs,
+  name,
+}: {
+  j: JSCodeshift;
+  expression: ArrowFunctionExpression | ObjectExpression;
+  attrs: ObjectProperty[];
+  name: string;
+}) => {
+  const propsParam =
+    expression.type === 'ArrowFunctionExpression' && expression.params[0].type === 'Identifier'
+      ? expression.params[0]
+      : j.identifier('props');
+  const jsxAttrs: (JSXAttribute | JSXSpreadAttribute)[] = attrs.map((attr) =>
+    j.jsxAttribute(
+      j.jsxIdentifier((attr.key as Identifier).name),
+      j.jsxExpressionContainer(attr.value as Parameters<typeof j.jsxExpressionContainer>[0])
+    )
+  );
+
+  jsxAttrs.push(j.jsxSpreadAttribute(propsParam));
+
+  const component = j.jsxElement(
+    j.jsxOpeningElement(j.jsxIdentifier(getAlternativeComponentName(name)), jsxAttrs, true)
+  );
+
+  return j.variableDeclaration('const', [
+    j.variableDeclarator(j.identifier(name), j.arrowFunctionExpression([propsParam], component)),
+  ]);
+};
+
+const createCompiledComponent = ({
+  j,
+  expression,
+  name,
+}: {
+  j: JSCodeshift;
+  expression: TaggedTemplateExpression;
+  name: string;
+}) => j.variableDeclaration('const', [j.variableDeclarator(j.identifier(name), expression)]);
+
+const applyBuildAttributes = ({
+  plugins,
+  originalNode,
+  transformedNode,
+  extraContent,
+}: {
+  plugins: Array<CodemodPluginInstance>;
+  originalNode: VariableDeclaration;
+  transformedNode: VariableDeclaration;
+  extraContent: VariableDeclaration | null;
+}) => {
+  plugins.reduce((currentNode, plugin) => {
+    const buildAttributesImpl = plugin.transform?.buildAttributes;
+    if (!buildAttributesImpl) {
+      return currentNode;
+    }
+
+    return buildAttributesImpl({
+      originalNode,
+      transformedNode,
+      currentNode,
+      extraContent,
+    });
+  }, originalNode as VariableDeclaration | ExportNamedDeclaration);
+};
+
+export const convertStyledAttrsToComponent = ({
+  j,
+  plugins,
+  templateExpressions,
+}: {
+  j: JSCodeshift;
+  plugins: CodemodPluginInstance[];
+  templateExpressions: Collection<TaggedTemplateExpression>;
+}): void => {
+  templateExpressions.forEach((templateExpression) => {
+    let newComponentDeclaration = null;
+    let extraContent = null;
+
+    const componentDeclarator = templateExpression.parentPath;
+    const componentDeclaration = componentDeclarator.parentPath.parentPath;
+    const componentName = componentDeclarator.value.id.name;
```

Comment:
nit: I think this would be easier to read if the variables were moved closer to their first usage